### DISCUSSION TITLE
ToggleButton LookAndFeel with Image

### DISCUSSION THREAD

**Discussion by dnaldoog** [RomanKubiak/ctrlr#640] [Show and tell]
Here is a simple example of how to use `LookAndFeel` to load an image into a `uiToggleButton`.

If you are in the Americas, just turn your phone or computer upside down .🙃 

I found it easier just to use a separate `uiLabel` instead of positioning text in `LookAndFeel`.

![image](https://github.com/RomanKubiak/ctrlr/assets/76831683/5339719b-9c78-4213-9495-b62d7a1d7d32)
----
![image](https://github.com/RomanKubiak/ctrlr/assets/76831683/03b098a5-357a-40f8-89ab-32330303156e)

----
[DEMO - uiToggleButton L&F.zip](https://github.com/user-attachments/files/23427189/DEMO.-.uiToggleButton.L.F.zip)

<pre>
lf = LookAndFeel()
lf:setImplementation (lookAndFeelImpl)
panel:getCanvas():setCustomLookAndFeel(lf)

lookAndFeelImpl = {
    drawToggleButton = function(g, toggleButton, isMouseOverButton, isButtonDown)
        if toggleButton:getToggleState() then -- ON
            local ton = resources:getResource("toggon")
            if ton ~= nil then
                ton = ton:asImage()
                g:drawImageAt(ton, 0, 0, false)
            end -- image found

        else -- OFF
            local toff = resources:getResource("toggoff")
            if toff ~= nil then
                toff = toff:asImage()
                g:drawImageAt(toff, 0, 0, false)
            end -- image found
        end
    end

} -- End table
</pre>

**Comment by unityconstruct** [comment]
> I found it easier just to use a separate `uiLabel` instead of positioning text in `LookAndFeel`.
Yeah, I have a helluva time getting the formatting with the component's label Justify/width/height settings  sometimes.

**nice tip** on the LnF Custom Image !

**Reply by dnaldoog** [reply]
Thanks @unityconstruct you're back!

**Reply by unityconstruct** [reply]
haha.. @dnaldoog ..  I wish!.. I jumped into Ctrlr for 18hr/day for 3mo when my contract ended on 1/1 after putting it off for 3-4yrs due to `not another f'ing tech stack`(C/Lua) so it was a "Now or Never" moment... I didn't even recognize my front yard when I came up for air & no sooner than I came up, the same firm pulled me back under. So while money is coming in, the incessant nagging of `The Incomplete Project` is unsufferable. Fortunately, I got about 80% of the EMU CS/P2K roughly working & trapped enough notes to be able pick up where I left off without the 90degree learning curve. 

`I felt bad actually, b/c I'm sure folks were wistfully checking here daily for another installment of my novel-length posts` 😂

**Reply by dnaldoog** [reply]
> I felt bad actually, b/c I'm sure folks were wistfully checking here daily for another installment of my novel-length posts 😂

Yes, checking in to learn new turns of phrase  etc ー the *incessant nagging* phenomenon is a thing for sure. 😺👻
